src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -162,18 +166,21 @@ SELECT COUNT(DISTINCT p)
     boolean existsByParticipation_ExerciseId(long exerciseId);
 
     /**
+     * Use this method only for exams!
      * Given an exerciseId and a correctionRound, return the number of assessments for that exerciseId and correctionRound that have been finished
      *
      * @param exercise  - the exercise we are interested in
-     * @param correctionRounds - the correction round we want finished assessments for
+     * @param numberOfCorrectionRounds - the correction round we want finished assessments for
      * @return an array of the number of assessments for the exercise for a given correction round
      */
-    default DueDateStat[] countNumberOfFinishedAssessmentsForExerciseForCorrectionRound(Exercise exercise, int correctionRounds) {
-        DueDateStat[] correctionRoundsDataStats = new DueDateStat[correctionRounds];
+    default DueDateStat[] countNumberOfFinishedAssessmentsForExamExerciseForCorrectionRound(Exercise exercise, int numberOfCorrectionRounds) {
+        DueDateStat[] correctionRoundsDataStats = new DueDateStat[numberOfCorrectionRounds];
+
+        List<Long> countlist = countNumberOfFinishedAssessmentsByExerciseIdIgnoreTestRuns(exercise.getId());
 
-        for (int i = 0; i < correctionRounds; i++) {
-            var finishedAssessments = this.countNumberOfFinishedAssessmentsByCorrectionRoundsAndExerciseIdIgnoreTestRuns(exercise.getId(), i);
-            correctionRoundsDataStats[i] = new DueDateStat(finishedAssessments, 0L);
+        correctionRoundsDataStats[0] = new DueDateStat(countlist.stream().filter(x -> x >= 1L).count(), 0L);
+        if (numberOfCorrectionRounds == 2) {
```

Comment:
What if numberOfCorrectionRounds is 3?

Reply:
at the moment this case can never happen, so it is fine

Reply:
Could you maybe add a comment for future developers?